### PR TITLE
Swarrot implementation in order to support PECL lib and videlalvaro (and graceful restart)

### PR DESCRIPTION
**Pull request by remi-san**
Full support of previous funcs.

See new examples.

### CODE REVIEW

File: examples/async-message-sender.php [hunk 1]
```diff
@@ -9,7 +9,26 @@
...
+$exchange = 'xchange';
+
+//$publisher = new \Burrow\RabbitMQ\AmqpAsyncPublisher($host, $port, $user, $pass, $exchange);
+
+// $connection = new \PhpAmqpLib\Connection\AMQPStreamConnection($host, $port, $user, $pass);
+// $channel = $connection->channel();
+// $messagePublisher = new \Swarrot\Broker\MessagePublisher\PhpAmqpLibMessagePublisher($channel, $exchange);
+
+$credentials = array('host' => $host, 'port' => $port, 'vhost' => '/', 'login' => $user, 'password' => $pass);
+$connection = new \AMQPConnection($credentials);
```
Comment (oliviermadre) [inline examples/async-message-sender.php]: this only works if you have PECL extension installed ?

Comment (remi-san) [inline examples/async-message-sender.php]: Yep.

If you want to use the swarrot implementation of burrow, you'll have to build your swarrot object with the implementation you want.

I might create a factory to retrieve the swarrot objetcs, but that won't be part of that lib.

Comment (oliviermadre) [inline examples/async-message-sender.php]: In this case, I'd rather see an example with and w/o Swarrot.
If I want to use Burrow without Swarrot, I have no more working examples.

**Comment by remi-san** [comment]
Won't merge

**Pull request closed without merging**